examples/basic.rs

```diff
@@ -1,34 +1,38 @@
-use async_imap::error::Result;
+use async_imap::error::{Error, Result};
 use async_std::prelude::*;
 use async_std::task;
+use std::env;
 
 fn main() -> Result<()> {
     task::block_on(async {
-        // To connect to the gmail IMAP server with this you will need to allow unsecure apps access.
-        // See: https://support.google.com/accounts/answer/6010255?hl=en
-        // Look at the gmail_oauth2.rs example on how to connect to a gmail server securely.
-        fetch_inbox_top().await?;
-        Ok(())
+        let args: Vec<String> = env::args().collect();
+        if args.len() != 4 {
+            eprintln!("need three arguments: imap-server login password");
+            Err(Error::Bad("need three arguments".into()))
```

Comment:
that's a bit of a weird error, didn't know how to do it better (i.e. have what is in the eprintln be returned and printed?).

Reply:
you can just `panic` if you want to